File: src/GiscusBlazor/Giscus.razor.js
export function addGiscus(repo,repoId,category,categoryId,mapping,reactionsEnabled,emitMetadata,inputPosition,theme,lang) {
	const giscus = document.createElement("script");
	giscus.setAttribute("src", "https://giscus.app/client.js");
	giscus.setAttribute("data-repo", repo);
	giscus.setAttribute("data-repo-id", repoId);
	giscus.setAttribute("data-category", category);
	giscus.setAttribute("data-category-id", categoryId);
	giscus.setAttribute("data-mapping", mapping);
	giscus.setAttribute("data-reactions-enabled", reactionsEnabled);
	giscus.setAttribute("data-emit-metadata", emitMetadata);
	giscus.setAttribute("data-input-position", inputPosition);
	giscus.setAttribute("data-theme", theme);
	giscus.setAttribute("data-lang", lang);
	giscus.setAttribute("crossorigin", "anonymous");
	giscus.setAttribute("async", "");
	document.getElementById("giscus-block").appendChild(giscus);
}
export function addGiscusWithTerm(repo, repoId, category, categoryId, mapping,term, reactionsEnabled, emitMetadata, inputPosition, theme, lang) {
	const giscus = document.createElement("script");
	giscus.setAttribute("src", "https://giscus.app/client.js");
	giscus.setAttribute("data-repo", repo);
	giscus.setAttribute("data-repo-id", repoId);
	giscus.setAttribute("data-category", category);
	giscus.setAttribute("data-category-id", categoryId);
	giscus.setAttribute("data-mapping", mapping);
	giscus.setAttribute("data-term",term);
	giscus.setAttribute("data-reactions-enabled", reactionsEnabled);
	giscus.setAttribute("data-emit-metadata", emitMetadata);
	giscus.setAttribute("data-input-position", inputPosition);
	giscus.setAttribute("data-theme", theme);
	giscus.setAttribute("data-lang", lang);
	giscus.setAttribute("crossorigin", "anonymous");
	giscus.setAttribute("async", "");
	document.getElementById("giscus-block").appendChild(giscus);
}

File: src/GiscusBlazor/Giscus.razor.cs
using Microsoft.AspNetCore.Components;

namespace GiscusBlazor
{
    public partial class Giscus
    {
        [EditorRequired]
        [Parameter]
        public string Repo { get; set; } = null!;

        [EditorRequired]
        [Parameter]
        public string RepoId { get; set; } = null!;

        [Parameter]
        public string? Category { get; set; }

        [Parameter]
        public string? CategoryId { get; set; }

        [EditorRequired]
        [Parameter]
        public Mapping Mapping { get; set; } = Mapping.PathName;

        [Parameter]
        public string? Term { get; set; }

        [Parameter]
        public bool Strict { get; set; } = false;

        [Parameter]
        public bool ReactionsEnabled { get; set; } = true;

        [Parameter]
        public bool EmitMetadata { get; set; }

        [Parameter]
        public InputPosition InputPosition { get; set; }

        [Parameter]
        public string? Theme { get; set; }

        [Parameter]
        public string? Lang { get; set; }

        [Parameter]
        public Loading Loading { get; set; } = Loading.Eager;

        [Parameter]
        public string? Host { get; set; } = "https://giscus.app";
    }
}
